src/init.cpp

```diff
@@ -1484,6 +1492,58 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
                     }
                 }
 
+                // Check -txoutindex
+                pcoinsByScriptDB->ReadFlag("txoutindex", fTxOutIndex);
+                if (IsArgSet("-txoutindex"))
+                {
+                    if (GetBoolArg("-txoutindex", false))
+                    {
+                        // build index
+                        if (!fTxOutIndex)
```

Comment:
We also need to build the index if it's outdated. For example, if the user ran an older version to sync. (This is fixed automatically if you replace the flag with a check that the index's best block matches that of the chainstate.)